packages/mobile/src/screens/collection-screen/CollectionScreenDetailsTile.tsx

```diff
@@ -409,127 +372,213 @@ export const CollectionScreenDetailsTile = ({
 
   return (
     <Paper mb='2xl' style={{ overflow: 'hidden' }}>
-      {renderDogEar()}
-      <Flex p='l' gap='l' alignItems='center' w='100%'>
-        <Text
-          variant='label'
-          size='m'
-          strength='default'
-          textTransform='uppercase'
-          color='subdued'
-        >
-          {messages.collectionType}
-        </Text>
+      <ScreenPrimaryContent>
+        {renderDogEar()}
+        <Flex p='l' gap='l' alignItems='center' w='100%'>
+          <Text
+            variant='label'
+            size='m'
+            strength='default'
+            textTransform='uppercase'
+            color='subdued'
+          >
+            {messages.collectionType}
+          </Text>
 
-        {badges.length > 0 ? (
-          <Flex direction='row' gap='s'>
-            {badges.map((badge) => badge)}
+          {badges.length > 0 ? (
+            <Flex direction='row' gap='s'>
+              {badges.map((badge) => badge)}
+            </Flex>
+          ) : null}
+          {imageElement}
+          <Flex gap='xs' alignItems='center'>
+            <Text variant='heading' size='s' textAlign='center'>
+              {title}
+            </Text>
+            {user ? (
+              <TouchableOpacity onPress={handlePressArtistName}>
+                <Flex direction='row' gap='xs'>
+                  <Text variant='body' color='accent' size='l'>
+                    {user.name}
+                  </Text>
+                  <UserBadges badgeSize={spacing.l} user={user} hideName />
+                </Flex>
+              </TouchableOpacity>
+            ) : null}
           </Flex>
-        ) : null}
-        {imageElement}
-        <Flex gap='xs' alignItems='center'>
-          <Text variant='heading' size='s' textAlign='center'>
-            {title}
-          </Text>
-          {user ? (
-            <TouchableOpacity onPress={handlePressArtistName}>
-              <Flex direction='row' gap='xs'>
-                <Text variant='body' color='accent' size='l'>
-                  {user.name}
-                </Text>
-                <UserBadges badgeSize={spacing.l} user={user} hideName />
-              </Flex>
-            </TouchableOpacity>
+          {shouldShowPlay ? (
+            <Button
+              iconLeft={isPlaying ? IconPause : IconPlay}
+              onPress={handlePressPlay}
+              disabled={!isPlayable}
+              fullWidth
+            >
+              {isPlaying ? messages.pause : messages.play}
+            </Button>
           ) : null}
+          {shouldShowPreview ? <PreviewButton /> : null}
+          <DetailsTileActionButtons
```

Comment:
did you notice that rendering all this lead to some speed issues or was it still okay to render all this in the "PrimaryScreenContent"?

Reply:
Unfortunately, I wasn't able to get a granular enough measurement for whether this was impacting it or not. I made a follow up ticket to push the callback creation down towards a wrapper around the buttons so we can have `<CollectionDetailsTileActionButtons collectionId={collectionId} />` which builds the callbacks and renders `<DetailsTileActionButtons />` as secondary content with a skeleton.

Not sure if it's worth the impact or not, and it's kind of nice to have the buttons appear on first render.